src/commands.rs

```diff
@@ -2882,6 +2882,31 @@ don't make any changes, then the operation will be aborted.",
     Ok(())
 }
 
+fn _description_template_for_cmd_split(
+    workspace_command: &WorkspaceCommandHelper,
+    intro: &str,
+    overall_commit_description: &str,
+    diff_iter: TreeDiffIterator,
+) -> Result<String, CommandError> {
+    let result1 = format!(
+        "JJ: {intro}\n{}\nJJ: This part contains the following changes:\n",
+        overall_commit_description,
+    );
+
+    let mut formatter = PlainTextFormatter::new(vec![]);
+    show_diff_summary(&mut formatter, workspace_command, diff_iter)?;
+
+    let mut result2 = String::default();
+    std::str::from_utf8(formatter.output())
+        .map(|r| result2.push_str(r))
+        .unwrap_or_else(|err| result2.push_str(&format!("Error displaying diff: {err}")));
```

Comment:
I take it, diffs are always valid UTF8? I didn't assume that because I could imagine some weird cases:

- Weird filenames that aren't valid UTF8. I guess you're saying these are not allowed in jj repos?

- If we ever include an option to show file content, and the files contain crazy content. I guess we don't show a diff in this case, even for long diffs?

Reply:
> * Weird filenames that aren't valid UTF8. I guess you're saying these are not allowed in jj repos?

Exactly.

> If we ever include an option to show file content, and the files contain crazy content. I guess we don't show a diff in this case, even for long diffs?

Never say never, but I expect the `diff_summary()` function used here to never include content diffs (we have other functions for that).